.ci/Jenkinsfile

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'metal || darwin || master' }
+  options {
+    timeout(time: 10)
+    buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  stages {
+    /**
+     This creates a GitHub comment linking to the preview of the docs build. The docs preview build itself is generated here:
+     https://elasticsearch-ci.elastic.co/view/Docs/job/elastic+observability-docs+pull-request+build-docs/
+    */
+    stage('Generate Docs Link') {
+      steps {
+        deleteDir()
+        script {
+            def PRnum = sh(script: "echo ${CHANGE_URL} | cut -d / -f7", returnStdout: true)
+            githubPrComment(message: "Docs preview will be available at: `https://observability-docs_${PRnum}.docs-preview.app.elstc.co/guide/en/observability/current/index.html`")
+        }
```

Comment:
NIT: you can skip the sh step by using `CHANGE_ID` that it is set with the PR id otherwise unset

Something like:
```suggestion
        githubPrComment(message: "Docs preview will be available at: `https://observability-docs_${CHANGE_ID}.docs-preview.app.elstc.co/guide/en/observability/current/index.html`")
```

but if so, then a when condition in the stage with only when a change request could avoid the issues to move to MBP for branches

```
when { changeRequest() }
```